src/state_machine/coordinator/fire.rs

```diff
@@ -1389,57 +1529,23 @@ pub mod test {
     }
 
     fn minimum_signers_sign<Aggregator: AggregatorTrait, Signer: SignerTrait>() {
-        let num_signers = 5;
+        let num_signers = 10;
         let keys_per_signer = 2;
+
         let (mut coordinators, mut signers) =
-            setup::<FireCoordinator<Aggregator>, Signer>(num_signers, keys_per_signer);
+            minimum_signers_dkg::<Aggregator, Signer>(num_signers, keys_per_signer);
         let config = coordinators.first().unwrap().get_config();
 
-        // We have started a dkg round
-        let message = coordinators.first_mut().unwrap().start_dkg_round().unwrap();
-        assert!(coordinators.first().unwrap().aggregate_public_key.is_none());
-        assert_eq!(coordinators.first().unwrap().state, State::DkgPublicGather);
-
-        // Send the DKG Begin message to all signers and gather responses by sharing with all other signers and coordinator
-        let (outbound_messages, operation_results) =
-            feedback_messages(&mut coordinators, &mut signers, &[message]);
-        assert!(operation_results.is_empty());
-        for coordinator in &coordinators {
-            assert_eq!(coordinator.state, State::DkgEndGather);
-        }
-
-        // Successfully got an Aggregate Public Key...
-        assert_eq!(outbound_messages.len(), 1);
-        match &outbound_messages[0].msg {
-            Message::DkgPrivateBegin(_) => {}
-            _ => {
-                panic!("Expected DkgPrivateBegin message");
-            }
-        }
-
-        // Send the DKG Private Begin message to all signers and share their responses with the coordinator and signers
-        let (outbound_messages, operation_results) =
-            feedback_messages(&mut coordinators, &mut signers, &outbound_messages);
-        assert!(outbound_messages.is_empty());
-        assert_eq!(operation_results.len(), 1);
-        match operation_results[0] {
-            OperationResult::Dkg(point) => {
-                assert_ne!(point, Point::default());
-                for coordinator in &coordinators {
-                    assert_eq!(coordinator.aggregate_public_key, Some(point));
-                    assert_eq!(coordinator.state, State::Idle);
-                }
-            }
-            _ => panic!("Expected Dkg Operation result"),
-        }
-
         // Figure out how many signers we can remove and still be above the threshold
         let num_keys = config.num_keys as f64;
         let threshold = config.threshold as f64;
-        let num_signers_to_remove =
+        let mut num_signers_to_remove =
             ((num_keys - threshold) / keys_per_signer as f64).floor() as usize;
+        if num_signers as usize > signers.len() {
+            num_signers_to_remove -= (num_signers - signers.len() as u32) as usize;
+        }
         for _ in 0..num_signers_to_remove {
```

Comment:
You should be able to just remove this no? Since you no longer are popping anything off the signers?

Reply:
I need to go back over this code.  I tried to reuse the `_dkg` test config/coordinators/signers in the `_sign` tests, but for the `minimum` and `insufficient` tests that meant that we'd already removed some signers from the list, and it caused problems to remove it again, so at some point I just stopped removing things to get the tests to work.  

But now that it's working as is I can tweak what gets removed for what stage of which test.

Reply:
Ok I uncommented the pop, so now we're testing with dkg threshold `9/10` and sign `7/10` as originally intended, and as we expect to run in production.  Fortunately it works now :100: